src/services/organizeImports.ts

```diff
@@ -164,7 +165,29 @@ namespace ts.OrganizeImports {
         function isDeclarationUsed(identifier: Identifier) {
             // The JSX factory symbol is always used if JSX elements are present - even if they are not allowed.
             return jsxElementsPresent && (identifier.text === jsxNamespace || jsxFragmentFactory && identifier.text === jsxFragmentFactory) && jsxModeNeedsExplicitImport(compilerOptions.jsx) ||
-                FindAllReferences.Core.isSymbolReferencedInFile(identifier, typeChecker, sourceFile);
+                FindAllReferences.Core.isSymbolReferencedInFile(identifier, typeChecker, sourceFile) ||

```

Comment:
organiseImports should rely on FindAllReferences to determine whether a link tag is referenced.
I'm not 100% sure how it should work, but I think that FindAllReferences, when determining whether `ZodType` is referenced, should

1. find the text `ZodType` in the text of an `@link` tag. (This should already be the case).
2. Use checker.getSymbolAtLocation to resolve the `link.name` to a symbol. I don't know what FindAllReferences normally uses to resolve names. You might have to add a special case here for when the identifier is a `link.name`.
3. If the name resolves, then it is referenced and `isSymbolReferencedInFile` should return true.

Reply:
I have a question about this: I saw that the find all references has two methods that work with JSDocs: `getParameterSymbolFromJSDoc` and `getTypeParameterFromJsDoc`. From what I could gather, both methods traverse the tree to find the appropriate symbol for a given JSDoc element.

For example, the `getParameterSymbolFromJSDoc` is used for `@param` tags and checks the function declaration that contains this JSDoc to get the Symbol from the function's arguments. 

Something similar happens with the `getTypeParameterFromJsDoc`  function (I think).

My point is that we are not creating new Symbols, but retrieving them from the existing elements of the tree and returning them.

However, in the case of my PR, we expect that the Symbol for the `{@link}` tag is on the top of the tree on the import declarations. So do I have to traverse all the way up the tree to get the correct Symbol, or do you think there is a better way to do this?